File: workflow_shell/git/checkout-new-branch.py
from ..util import run


command_string = "gcb"
command_help = "Takes one argument, the name of the branch. Checks out a new branch"
short_help = "Git - checks out a new branch"
command_arguments = ["branch_name"]
argument_required = True


def main(*args, **kwargs):
    argument = kwargs.get(command_arguments[0])
    run(["git", "checkout", "-b", argument])

File: src/commands/git/CheckoutNewBranch.ts
import { Option } from '../../common/interface/Option';
import { SuperCommand } from '../../common/SuperCommand';
import { executeCommand } from '../../common/Cli';
import { Argument } from '../../common/interface/Argument';
import { Names } from '../../common/interface/Names';

const argumentss: Argument[] = [];
const options: Option[] = [];

export class Command extends SuperCommand {
  description: string = 'Takes one argument, the name of the branch. Checks out a new branch.';
  alias: string = 'gcb';
  public async execute(vital?: boolean, input?: any) {
    await super.executeWithInput(argumentss, options, input, vital, async () => {
      await executeCommand('git', ['checkout', '-b', this.input[Names.BRANCH]]);
    });
  }
}
